### PR TITLE
Refactor SDA tests into unit tests

### PR DESCRIPTION
**Pull request by nikkolasg**
Refactor the tests in SDA to have unit tests *inside the package* (`package sda` vs `package sda_test`). 
See issue #430 . 
**NOTE**: Since we are trying to have a clean / organized way of working, I decided to let this PR as is, meaning I *won't try* to reduce the exported API of sda *here*, I will do it in a separate issue since this will most likely generate a lot of discussion (*what should be exported & what should not*).

This PR includes the changes of #468 (the commits will disappear when #468 will be merged)
=> Preferable to merge #468  first.

### CODE REVIEW

**Comment by nikkolasg** [comment]
Here's all the exported function that I listed in the test in the `sda/` package.
[export_sda.txt](https://github.com/dedis/cothority/files/366090/export_sda.txt)

File: sda/node_test.go [hunk 1]
```diff
@@ -265,57 +263,61 @@ func TestFlags(t *testing.T) {
...
+//}
+//// XXX using manage here ... maybe theses are the kind of protocol /
+//// services we want to embed directly into SDA instead of exporting them and
+//// then importing them again in sda_test. If we really need them ...
+//protoCount := root.(*manage.ProtocolCount)
+//count := <-protoCount.Count
+//if count != 7 {
+//t.Fatal("Didn't get a count of 7:", count)
+//}
+/*}*/
```
Comment (Liamsi) [inline sda/node_test.go]: I think this could be removed.

**Comment by Liamsi** [comment]
OK to me

**Comment by ineiti** [comment]
Can you make that depend on #468 for easier comparison of the files?

**Comment by Liamsi** [comment]
For simpler reviewing (as long as #468 isn't merged): https://github.com/dedis/cothority/compare/refactor_processor...refactor_unit

**Comment by Liamsi** [comment]
Looks good to me! Will merge after #468

**Pull request merged**